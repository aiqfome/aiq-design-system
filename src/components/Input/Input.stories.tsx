import React from 'react'
import { withKnobs, text, select, boolean } from '@storybook/addon-knobs'

import { MdSearch } from 'react-icons/md'

import { Flex } from '../Flex'
import { Input } from './Input'

export default {
  component: Input,
  title: 'Input',
  decorators: [withKnobs]
}

export const Basic: React.FC = () => (
  <Flex variant='fullCentralized'>
    <Input
      value={text('value', '')}
      errorForm={boolean('errorForm', false)}
      errorMessage={text('errorMessage', 'message error')}
      label={text('label', 'aiqfome')}
      placeholder={text('placeholder', 'duas pizzas é muito')}
      type={select(
        'Type',
        {
          Text: 'text',
          Password: 'password'
        },
        'text'
      )}
    />
  </Flex>
)

export const Outlined: React.FC = () => (
  <Flex variant='fullCentralized'>
    <Input
      variant='outlined'
      value={text('value', '')}
      errorForm={boolean('errorForm', false)}
      errorMessage={text('errorMessage', 'message error')}
      label={text('label', 'aiqfome')}
      type={select(
        'Type',
        {
          Text: 'text',
          Password: 'password'
        },
        'text'
      )}
    />
  </Flex>
)

export const Disabled: React.FC = () => (
  <Flex variant='fullCentralized'>
    <Input
      value={text('value', '')}
      errorForm={boolean('errorForm', false)}
      errorMessage={text('errorMessage', 'message error')}
      label={text('label', 'aiqfome')}
      placeholder={text('placeholder', 'duas pizzas é muito')}
      disabled
      variant={select(
        'Variant',
        {
          Outlined: 'outlined',
          Default: 'default'
        },
        'outlined'
      )}
    />
  </Flex>
)

export const Password: React.FC = () => (
  <Flex variant='fullCentralized'>
    <Input label='Aiqfome' type='password' />
  </Flex>
)

export const ErrorMessage: React.FC = () => (
  <Flex variant='fullCentralized'>
    <Input label='Aiqfome' errorForm={true} errorMessage='Not Allowed' />
  </Flex>
)

export const WithValue: React.FC = () => {
  const value = 'hamburger'

  return (
    <Flex variant='fullCentralized'>
      <Input label='Aiqfome' value={value} />
    </Flex>
  )
}

export const Sufix: React.FC = () => (
  <Flex variant='fullCentralized'>
    <Input label='Aiqfome' sufix={<MdSearch />} />
  </Flex>
)

export const Prefix: React.FC = () => (
  <Flex variant='fullCentralized'>
    <Input label='Aiqfome' prefix={<MdSearch color='#BABCBE' />} />
  </Flex>
)

export const Styled: React.FC = () => (
  <Flex variant='fullCentralized'>
    <Input
      width='530px'
      border='none'
      placeholder='procure aqui alguma coisa desse painelzão da p*$%#'
      backgroundColor='#F8F8F8'
      label='Aiqfome'
      prefix={<MdSearch color='#BABCBE' />}
    />
  </Flex>
)
