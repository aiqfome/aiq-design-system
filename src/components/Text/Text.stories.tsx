import React from 'react'
import { withKnobs, text, number } from '@storybook/addon-knobs'

import { Text } from './Text'
import { Flex } from '../Flex'

export default {
  component: Text,
  title: 'Text',
  decorators: [withKnobs]
}

export const basic: React.FC = () => (
  <Text
    color={text('color', 'primary')}
    fontSize={number('fontSize', 16)}
    cursor={text('cursor', '')}
    fontWeight={text('fontWeight', 'medium')}
  >
    {text('text', 'Aiqfome')}
  </Text>
)

export const cursor: React.FC = () => (
  <Text cursor='pointer'>Design System</Text>
)

export const WhiteSpace: React.FC = () => (
  <Flex width='50px' border='1px solid black'>
    <Text whiteSpace='nowrap'>Design System</Text>
  </Flex>
)
