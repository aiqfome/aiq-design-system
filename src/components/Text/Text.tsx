import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import {
  color,
  ColorProps,
  space,
  SpaceProps,
  layout,
  LayoutProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps,
  typography,
  TypographyProps
} from 'styled-system'

export interface Props
  extends ColorProps,
    SpaceProps,
    LayoutProps,
    FontSizeProps,
    FontWeightProps,
    TypographyProps {
  cursor?: string
  whiteSpace?: 'nowrap' | 'normal' | 'pre'
}

export const Text = styled.span<Props>`
  ${color}
  ${space}
  ${layout}
  ${fontSize}
  ${fontWeight}
  ${typography}

  ${({ cursor }) =>
    cursor &&
    css`
      cursor: ${cursor};
    `};
  white-space: ${props => props.whiteSpace || 'normal'};
`

Text.propTypes = {
  cursor: PropTypes.string,
  whiteSpace: PropTypes.oneOf(['nowrap', 'normal', 'pre'])
}
