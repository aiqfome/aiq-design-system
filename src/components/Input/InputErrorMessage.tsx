import React from 'react'

import { Text } from '../Text'

interface Props {
  errorForm?: boolean
  errorMessage?: string
}

export const InputErrorMessage: React.FC<Props> = ({ errorMessage }: Props) => {
  return (
    <Text color='grey' fontSize='small' mt={2}>
      {errorMessage}
    </Text>
  )
}
