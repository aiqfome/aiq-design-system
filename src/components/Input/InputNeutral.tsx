import React, { useState, InputHTMLAttributes } from 'react'
import PropTypes from 'prop-types'
import { MdVisibility, MdVisibilityOff } from 'react-icons/md'

import styled, { css } from 'styled-components'
import { space, layout, border, color } from 'styled-system'

import { Button } from '../Button'
import { Flex } from '../Flex'
import { Box } from '../Box'

import { InputErrorMessage } from './InputErrorMessage'

export interface Props extends InputHTMLAttributes<HTMLInputElement> {
  name?: string
  inputRef?: any
  errorForm?: boolean
  type?: string
  errorMessage?: string
  value?: string
  sufix?: any
  prefix?: any
  placeholder?: string

  backgroundColor?: any
  border?: any
  width?: any
  maxWidth?: any
}

const InputStyled = styled.input<Props>`
  ${space}
  ${layout}
  ${color}

  padding: 10px 12px;
  border: 1px solid ${({ theme }) => theme.colors.mediumGrey};
  border-radius: 4px;

  &:focus {
    border-color: ${({ theme }) => theme.colors.primary};
  }

  ${({ theme, errorForm }) =>
    errorForm &&
    css`
      border-color: ${theme.colors.error};
    `};
`

export interface PropsContainerSufix {
  inputSelected?: boolean
  errorForm?: boolean
  onClick?: () => void
  onBlur?: () => void
}

const ContainerSufix = styled(Box)<PropsContainerSufix>`
  display: flex;
  align-items: center;

  padding: 10px 12px;
  border: 1px solid ${({ theme }) => theme.colors.mediumGrey};
  border-radius: 4px;

  ${({ inputSelected }) =>
    inputSelected &&
    css`
      border-color: ${({ theme }) => theme.colors.primary};

      svg {
        color: ${({ theme }) => theme.colors.primary};
      }
    `}

  ${({ theme, errorForm }) =>
    errorForm &&
    css`
      border-color: ${theme.colors.error};
    `};
  ${border}
`

const InputSufixed = styled.input<Props>`
  ${layout}
  ${color}

  border: none;
`

const InputPrefixed = styled.input<Props>`
  ${layout}
  ${color}
  
  border: none;
  margin-left: 16px;
`

const Input: React.FC<Props> = ({ inputRef, value, ...props }) => {
  return <InputStyled value={value} ref={inputRef} {...props} />
}

export const InputNeutral: React.FC<Props> = ({
  name,
  inputRef,
  errorForm,
  errorMessage,
  type = 'text',
  sufix,
  prefix,
  value,
  placeholder,
  ...props
}) => {
  const [inputSelected, setInputSelected] = useState(false)
  const [passwordVisible, setPasswordVisible] = useState(false)

  const { backgroundColor, border, width, maxWidth } = props
  const boxStyled = {
    backgroundColor,
    border,
    width,
    maxWidth
  }

  if (sufix) {
    return (
      <Flex flexDirection='column'>
        <ContainerSufix
          {...boxStyled}
          inputSelected={inputSelected}
          onClick={() => setInputSelected(true)}
          onBlur={() => setInputSelected(false)}
        >
          <InputSufixed
            ref={inputRef}
            placeholder={placeholder}
            type={type}
            value={value}
            {...props}
          />
          {sufix}
        </ContainerSufix>

        {errorForm && <InputErrorMessage errorMessage={errorMessage} />}
      </Flex>
    )
  }

  if (prefix) {
    return (
      <Flex>
        <ContainerSufix
          {...boxStyled}
          inputSelected={inputSelected}
          onClick={() => setInputSelected(true)}
          onBlur={() => setInputSelected(false)}
        >
          {prefix}
          <InputPrefixed
            ref={inputRef}
            placeholder={placeholder}
            type={type}
            value={value}
            {...props}
          />
        </ContainerSufix>

        {errorForm && <InputErrorMessage errorMessage={errorMessage} />}
      </Flex>
    )
  }

  if (type === 'password') {
    return (
      <Flex flexDirection='column'>
        <ContainerSufix
          {...boxStyled}
          inputSelected={inputSelected}
          onClick={() => setInputSelected(true)}
          onBlur={() => setInputSelected(false)}
        >
          <InputSufixed
            ref={inputRef}
            placeholder={placeholder}
            type={passwordVisible ? 'text' : 'password'}
            value={value}
          />
          <Button
            palette='primary'
            mr={5}
            onClick={() => setPasswordVisible(!passwordVisible)}
          >
            {passwordVisible ? (
              <MdVisibilityOff size={22} />
            ) : (
              <MdVisibility size={22} />
            )}
          </Button>
        </ContainerSufix>

        {errorForm && <InputErrorMessage errorMessage={errorMessage} />}
      </Flex>
    )
  }

  return (
    <Flex flexDirection='column'>
      <Input
        name={name}
        inputRef={inputRef}
        placeholder={placeholder}
        errorForm={errorForm}
        type={type}
        errorMessage={errorMessage}
        sufix={sufix}
        value={value}
        {...props}
      />

      {errorForm && <InputErrorMessage errorMessage={errorMessage} />}
    </Flex>
  )
}

Input.propTypes = {
  inputRef: PropTypes.any,
  value: PropTypes.string
}

InputNeutral.propTypes = {
  name: PropTypes.string,
  inputRef: PropTypes.any,
  errorForm: PropTypes.bool,
  type: PropTypes.string,
  errorMessage: PropTypes.string,
  sufix: PropTypes.any,
  prefix: PropTypes.any,
  value: PropTypes.string,
  placeholder: PropTypes.string,

  backgroundColor: PropTypes.any,
  border: PropTypes.any,
  width: PropTypes.any,
  maxWidth: PropTypes.any
}
