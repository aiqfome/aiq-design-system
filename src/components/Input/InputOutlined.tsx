import React, { useState, InputHTMLAttributes } from 'react'
import PropTypes from 'prop-types'
import { MdVisibility, MdVisibilityOff } from 'react-icons/md'

import styled from 'styled-components'
import {
  color,
  space,
  SpaceProps,
  layout,
  fontSize,
  fontWeight
} from 'styled-system'

import { Button } from '../Button'
import { Text } from '../Text'
import { Box } from '../Box'

import { InputErrorMessage } from './InputErrorMessage'

export interface Props
  extends InputHTMLAttributes<HTMLInputElement>,
    SpaceProps {
  name?: string
  inputRef?: any
  label?: string
  errorForm?: boolean
  type?: string
  errorMessage?: string
  value?: string
  sufix?: any

  maxWidth?: number | string
  backgroundColor?: number | string
}

const Container = styled(Box)`
  ${color}
  ${space}
  ${layout}
  ${fontSize}
  ${fontWeight}
`

const LabelStyled = styled.label<Props>`
  position: relative;
  top: -3px;
  padding-top: 6px;
  line-height: 1.5;
  overflow: hidden;
  display: flex;
  align-items: center;

  & > input {
    max-height: 37px;
    box-sizing: border-box;
    margin: 0;
    border: solid 1px
      ${({ theme, errorForm }) =>
        errorForm ? theme.colors.error : theme.colors.mediumGrey};
    border-top-color: transparent;
    border-radius: 4px;
    padding: 9px 13px 9px;
    caret-color: ${({ theme }) => theme.colors.almostBlack};
    color: ${({ theme }) => theme.colors.almostBlack};
    width: 100%;
    height: inherit;
    box-shadow: none;
    line-height: inherit;
    transition: border 0.2s, box-shadow 0.2s;
    background: ${({ theme, disabled }) =>
      disabled ? theme.colors.lightGrey : theme.colors.white};

    &:not(:focus):placeholder-shown {
      border-top-color: ${({ theme, errorForm }) =>
        errorForm ? theme.colors.error : theme.colors.mediumGrey};
    }

    &:not(:focus):placeholder-shown + span {
      font-size: inherit;
      line-height: 53px;
    }

    &:focus {
      border-color: ${({ theme }) => theme.colors.primary};
      border-top-color: transparent;
      box-shadow: inset 1px 0 ${({ theme }) => theme.colors.primaryLight},
        inset -1px 0 ${({ theme }) => theme.colors.primaryLight},
        inset 0 -1px ${({ theme }) => theme.colors.primaryLight};
      outline: none;
    }

    &:focus + span {
      color: ${({ theme }) => theme.colors.primary};
    }

    &:focus + span::before,
    &:focus + span::after {
      border-top-color: ${({ theme }) => theme.colors.primary};
      box-shadow: inset 0 1px ${({ theme }) => theme.colors.primaryLight};
    }

    & + span {
      position: absolute;
      top: 0;
      left: 0;
      display: flex;
      width: 100%;
      max-height: 100%;
      font-size: 75%;
      line-height: 15px;
      cursor: text;
      transition: color 0.2s, font-size 0.2s, line-height 0.2s;
      color: ${({ theme }) => theme.colors.grey};

      &::before,
      &::after {
        content: '';
        display: block;
        box-sizing: border-box;
        margin-top: 6px;
        border-top: solid 1px;
        min-width: 10px;
        height: 8px;
        pointer-events: none;
        box-shadow: inset 0 1px transparent;
        transition: border-color 0.2s, box-shadow 0.2s;
        border-top-color: ${({ theme, errorForm }) =>
          errorForm ? theme.colors.error : theme.colors.mediumGrey};
      }

      &::before {
        margin-right: 4px;
        border-left: solid 1px transparent;
        border-radius: 4px 0;
      }

      &::after {
        flex-grow: 1;
        margin-left: 4px;
        border-right: solid 1px transparent;
        border-radius: 0 4px;
      }
    }
  }

  button {
    right: 0;
    position: absolute;
    background: none;
    border: none;
  }
`

export const InputOutlined: React.FC<Props> = ({
  name,
  inputRef,
  label,
  errorForm,
  type = 'text',
  errorMessage,
  sufix,
  value,
  backgroundColor,
  maxWidth,
  marginRight,
  marginLeft,

  ...props
}) => {
  const [showPassword, setShowPassword] = useState(false)

  const styledContainer = {
    backgroundColor,
    maxWidth,
    marginRight,
    marginLeft
  }

  if (type === 'password') {
    return (
      <Container {...styledContainer} {...props}>
        <LabelStyled errorForm={errorForm}>
          <input
            {...props}
            placeholder=' '
            type={showPassword ? 'text' : 'password'}
            ref={inputRef}
            name={name}
          />
          <Text>{label}</Text>

          <Button
            palette='primary'
            mr={5}
            onClick={() => setShowPassword(!showPassword)}
          >
            {showPassword ? (
              <MdVisibilityOff size={22} />
            ) : (
              <MdVisibility size={22} />
            )}
          </Button>
        </LabelStyled>

        {errorForm && <InputErrorMessage errorMessage={errorMessage} />}
      </Container>
    )
  }

  if (sufix) {
    return (
      <Container {...styledContainer} {...props}>
        <LabelStyled errorForm={errorForm}>
          <input
            {...props}
            placeholder=' '
            type={type}
            value={value}
            ref={inputRef}
            name={name}
          />
          <Text>{label}</Text>

          {sufix}
        </LabelStyled>

        {errorForm && <InputErrorMessage errorMessage={errorMessage} />}
      </Container>
    )
  }

  return (
    <Container {...styledContainer} {...props}>
      <LabelStyled errorForm={errorForm}>
        <input
          {...props}
          value={value}
          placeholder=' '
          name={name}
          type={type}
          ref={inputRef}
          autoComplete='off'
        />
        <Text>{label}</Text>
      </LabelStyled>

      {errorForm && <InputErrorMessage errorMessage={errorMessage} />}
    </Container>
  )
}

InputOutlined.propTypes = {
  name: PropTypes.string,
  inputRef: PropTypes.any,
  label: PropTypes.string,
  errorForm: PropTypes.bool,
  type: PropTypes.string,
  errorMessage: PropTypes.string,
  sufix: PropTypes.any,
  value: PropTypes.string,

  backgroundColor: PropTypes.any,
  maxWidth: PropTypes.any,
  marginLeft: PropTypes.any,
  marginRight: PropTypes.any
}
