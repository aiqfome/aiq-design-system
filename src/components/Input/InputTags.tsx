import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { MdClose } from 'react-icons/md'

import styled, { css } from 'styled-components'
import { space, layout, border, color } from 'styled-system'

import { Button } from '../Button'
import { Box } from '../Box'
import { Text } from '../Text'

import { InputErrorMessage } from './InputErrorMessage'

export interface Props {
  name?: string
  errorForm?: boolean
  type?: string
  errorMessage?: string
  value?: string
  placeholder?: string
  onChange?: (value: any) => void

  backgroundColor?: any
  border?: any
  width?: any
  maxWidth?: any
}

const InputStyled = styled.input<Props>`
  ${space}
  ${layout}
  ${color}
  border: none;
`

const Tag = styled.div`
  margin-right: 8px;
  background: ${({ theme }) => theme.colors.primary};
  border-radius: 4px;
  padding: 2px;
  display: flex;
  flex-direction: row;
`

export interface PropsContainerInput {
  inputSelected?: boolean
  errorForm?: boolean
  onClick?: () => void
  onBlur?: () => void
}

const ContainerInput = styled(Box)<PropsContainerInput>`
  display: flex;
  max-height: 37px;
  align-items: center;
  flex-direction: row;
  padding: 10px 12px;
  border: 1px solid ${({ theme }) => theme.colors.mediumGrey};
  border-radius: 4px;

  ${({ inputSelected }) =>
    inputSelected &&
    css`
      border-color: ${({ theme }) => theme.colors.primary};

      svg {
        color: ${({ theme }) => theme.colors.primary};
      }
    `}

  ${({ theme, errorForm }) =>
    errorForm &&
    css`
      border-color: ${theme.colors.error};
    `};
  ${border}
`

export const InputTags: React.FC<Props> = ({
  errorForm,
  errorMessage,
  type = 'text',
  placeholder,
  backgroundColor,
  border,
  width,
  maxWidth,
  value,
  onChange,
  ...props
}) => {
  const [inputValue, setInputValue] = useState<string>('')
  const [inputSelected, setInputSelected] = useState(false)
  const [tags, setTags] = useState<string[]>(value ? value.split(',') : [])

  const boxStyled = {
    backgroundColor,
    border,
    width,
    maxWidth
  }

  function addTag() {
    setTags([...tags, inputValue])
    setInputValue('')
    if (onChange) {
      onChange(tags)
    }
  }

  function removeTag(index) {
    const newTags = tags
    newTags.splice(index, 1)
    setTags([...newTags])
    if (onChange) {
      onChange(tags)
    }
  }

  function handleInputKeyDown(e) {
    if (e.keyCode === 13) {
      addTag()
    }
  }

  return (
    <ContainerInput
      inputSelected={inputSelected}
      onClick={() => setInputSelected(true)}
      onBlur={() => setInputSelected(false)}
      flexDirection='column'
      {...boxStyled}
    >
      <input type='hidden' value={tags.toString()} {...props} />
      <Box display='flex'>
        {tags.map((tag, index) => (
          <Tag key={index}>
            <Text fontSize='small' mr={1} color='white'>
              {tag}
            </Text>
            <Button onClick={() => removeTag(index)}>
              <MdClose color='#fff' size={12} />
            </Button>
          </Tag>
        ))}
      </Box>
      <InputStyled
        onKeyDown={handleInputKeyDown}
        placeholder={placeholder}
        type={type}
        onChange={e => setInputValue(e.target.value)}
        value={inputValue}
        {...boxStyled}
      />

      {errorForm && <InputErrorMessage errorMessage={errorMessage} />}
    </ContainerInput>
  )
}

InputTags.propTypes = {
  name: PropTypes.string,

  errorForm: PropTypes.bool,
  type: PropTypes.string,
  errorMessage: PropTypes.string,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
  backgroundColor: PropTypes.any,
  border: PropTypes.any,
  width: PropTypes.any,
  maxWidth: PropTypes.any
}
