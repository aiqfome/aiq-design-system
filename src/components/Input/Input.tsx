import React, { InputHTMLAttributes } from 'react'
import PropTypes from 'prop-types'

import { InputOutlined } from './InputOutlined'
import { InputNeutral } from './InputNeutral'

export interface Props extends InputHTMLAttributes<HTMLInputElement> {
  name?: string
  inputRef?: any
  label?: string
  errorForm?: boolean
  type?: string
  errorMessage?: string
  value?: string
  sufix?: any
  prefix?: any
  variant?: string
  placeholder?: string

  backgroundColor?: any
  border?: any
  width?: any
  maxWidth?: any
}

export const Input: React.FC<Props> = ({
  name,
  inputRef,
  label,
  errorForm,
  type = 'text',
  errorMessage,
  sufix,
  prefix,
  value,
  variant,
  placeholder,
  ...props
}) => {
  if (variant === 'outlined') {
    return (
      <InputOutlined
        name={name}
        inputRef={inputRef}
        label={label}
        errorForm={errorForm}
        type={type}
        errorMessage={errorMessage}
        sufix={sufix}
        value={value}
        {...props}
      />
    )
  }

  return (
    <InputNeutral
      name={name}
      inputRef={inputRef}
      errorForm={errorForm}
      type={type}
      errorMessage={errorMessage}
      sufix={sufix}
      prefix={prefix}
      value={value}
      placeholder={placeholder}
      {...props}
    />
  )
}

Input.propTypes = {
  name: PropTypes.string,
  inputRef: PropTypes.any,
  label: PropTypes.string,
  errorForm: PropTypes.bool,
  type: PropTypes.string,
  errorMessage: PropTypes.string,
  sufix: PropTypes.any,
  prefix: PropTypes.any,
  value: PropTypes.string,
  variant: PropTypes.string,
  placeholder: PropTypes.string,

  backgroundColor: PropTypes.any,
  border: PropTypes.any,
  width: PropTypes.any,
  maxWidth: PropTypes.any
}
