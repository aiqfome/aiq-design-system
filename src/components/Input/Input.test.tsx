import React from 'react'

import { Input } from './Input'
import { Text } from '../Text'

import { render } from '../utils/test/render'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(
      <Input>
        <Text>Input component!</Text>
      </Input>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
