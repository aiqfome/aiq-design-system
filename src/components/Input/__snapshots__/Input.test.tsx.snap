// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-AxjAm hgoUZd"
>
  <input
    className="sc-Axmtr lRNal"
    type="text"
  >
    <span
      className="sc-AxiKw bBViMC"
    >
      Input component!
    </span>
  </input>
</div>
`;
